htdocs/modules/system/class/maintenance.php

```
@@ -166,10 +172,43 @@ public function CleanAvatar()
+            //   - normalise backslashes to '/' (Windows-historic data)
+            //   - strip leading slashes (defends against absolute paths
+            //     accidentally or maliciously stored in the column)
+            //   - resolve via realpath() so any '../' segments collapse
+            //   - confirm the resolved path is contained inside the
+            //     resolved upload-root (prefix check with a trailing
+            //     separator so 'uploadsX/...' doesn't satisfy 'uploads')
+            //   - confirm it's a regular file before removal.
+            // Only then invoke the cleanup helper.
+            $avatarFile = ltrim(str_replace('\\', '/', (string) ($myrow['avatar_file'] ?? '')), '/');
+            if ('' === $avatarFile) {
+                $result1 = $this->db->exec('DELETE FROM ' . $this->db->prefix('avatar') . ' WHERE avatar_id=' . $myrow['avatar_id']);
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$result1'. (undefined)

(UnusedLocalVariable)